### PR TITLE
test: Cypress | Flaky fix SetOptions_Spec.ts for EE failure

### PR DESCRIPTION
**Pull request by Aishwarya-U-R**
## Description
- This PR fixes SetOptions_Spec.ts which is flaky in EE

#### Type of change
- Script fix (non-breaking change which fixes an issue)

#### How Has This Been Tested?
- [X] Cypress CI runs

#### QA activity:
- [X] Added `Test Plan Approved` label after Cypress tests were reviewed

## Summary by CodeRabbit

- **Tests**
  - Improved wait conditions and element retrieval methods for enhanced end-to-end test reliability.
  - Updated assertions to improve test accuracy and stability.

### CODE REVIEW

**Comment by Aishwarya-U-R** [comment]
EE passed here [Run1](https://github.com/appsmithorg/appsmith-ee/pull/3072#issuecomment-1865108332), [Run2](https://github.com/appsmithorg/appsmith-ee/pull/3072#issuecomment-1865126119)

File: app/client/cypress/e2e/Regression/ClientSide/SetProperty/SetOptions_Spec.ts [hunk 1]
```diff
@@ -299,11 +299,17 @@ describe(
     }`);
       jsEditor.EnableDisableAsyncFuncSettings("myFunc1", true, false); //for on page load execution, since sync function is updated to async
       deployMode.DeployApp();
-      agHelper.WaitUntilEleAppear(
-        locators._widgetInDeployed(draggableWidgets.INPUT_V2) +
-          " " +
-          locators._input,
-      );
+      cy.waitUntil(() => {
```
Comment (sarojsarab) [inline app/client/cypress/e2e/Regression/ClientSide/SetProperty/SetOptions_Spec.ts]: Please add max timeout for this wait.

File: app/client/cypress/e2e/Regression/ClientSide/SetProperty/SetOptions_Spec.ts [hunk 2]
```diff
@@ -340,18 +346,17 @@ describe(
       EditorNavigation.SelectEntityByName("Input1", EntityType.Widget);
       propPane.UpdatePropertyFieldValue("Default value", "{{Select3.options}}");
       deployMode.DeployApp();
-      cy.waitUntil(() =>
-        agHelper
-          .GetText(
+      cy.waitUntil(() => {
```
Comment (sarojsarab) [inline app/client/cypress/e2e/Regression/ClientSide/SetProperty/SetOptions_Spec.ts]: same here

**Review by sarojsarab — DISMISSED** [review]

File: app/client/cypress/e2e/Regression/ClientSide/SetProperty/SetOptions_Spec.ts (hunk 1 above)
Comment (Aishwarya-U-R) [reply app/client/cypress/e2e/Regression/ClientSide/SetProperty/SetOptions_Spec.ts]: Done

File: app/client/cypress/e2e/Regression/ClientSide/SetProperty/SetOptions_Spec.ts (hunk 2 above)
Comment (Aishwarya-U-R) [reply app/client/cypress/e2e/Regression/ClientSide/SetProperty/SetOptions_Spec.ts]: Thankyou, done

**Comment by Aishwarya-U-R** [comment]
@sarojsarab - can you please approve if if looks ok now?

**Review by sarojsarab — APPROVED** [review]
Looks good to me

**Pull request merged**